mip/mip.c

```diff
@@ -888,6 +896,14 @@ size_t mip_driver_rx(void *buf, size_t len, struct mip_if *ifp) {
 }
 
 void mip_init(struct mg_mgr *mgr, struct mip_if *ifp) {
+  // If MAC address is not set, make a random one
+  if (ifp->mac[0] == 0 && ifp->mac[1] == 0 && ifp->mac[2] == 0 &&
+      ifp->mac[3] == 0 && ifp->mac[4] == 0 && ifp->mac[5] == 0) {
+    mg_random(ifp->mac, sizeof(ifp->mac));
+    ifp->mac[0] &= (uint8_t) ~1;  // 1st byte must be even (unicast)
```

Comment:
How does this make it even ? There are other restrictions too; what about setting it to 02 that works for sure ?